exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerUtil.java

```diff
@@ -185,12 +248,173 @@ public static int sizeRepeatedMessage(
     return size;
   }
 
+  /** Returns the size of a repeated message field. */
+  public static <T> int sizeRepeatedMessage(
```

Comment:
Its a little surprising that these methods which are named like stateless pure functions have side affects on `MarshalerContext`. What do you think about breaking them into a separate `StatelessMarshalerUtil` to differentiate from the existing stateless utils? Could also name each method in a way which implies side affects on `MarshalerContext`. Something like `sizeRepeatedMessageWithContext`.

Would also be good to add javadoc link the util methods which are used in tandem to size / serialize. E.g. `sizeRepeatedMessage` is used with `serializeRepeatedMessage` since `serializeRepeatedMessage` relies on `sizeRepeatedMessage` updating the marshaler context with the size it computes.

This code is always going to be a bit hard to read / reason over, but we can give ourselves some bread crumbs to make it less brittle / error prone.